### ISSUE TITLE
[Feature]: `app dev` should update URLs and respect paths

### ISSUE THREAD

**Issue by zirkelc** [Shopify/cli#4219]
### What area(s) will this request affect?

App, Running your code locally

### What type of change do you want to see?

New feature

### Overview

When `shopify app dev` runs, it creates a tunnel via Cloudfare and updates the app URLs in Partners Dashboard. 

The URLs have the following format:
```toml
application_url = "https://trash-shopping-health-inspection.trycloudflare.com"

[auth]
redirect_urls = [
  "https://trash-shopping-health-inspection.trycloudflare.com/auth/callback",
  "https://trash-shopping-health-inspection.trycloudflare.com/auth/shopify/callback",
  "https://trash-shopping-health-inspection.trycloudflare.com/api/auth/callback"
]
```

where the subdomain `trash-shopping-health-inspection` is a random string that changes on every `app dev`. 

I would like to suggest to make it possible to configure the paths of each URL. For example, instead of `/auth/callback`  it should be possible to use `/app/auth/callback` or any other paths. Right now it's always `/` for the `application_url` and `/auth/callback`, `/auth/shopify/callback` and `/api/auth/callback` for `redirect_urls`.

It would be nice if there were two CLI options like `--app-url-path=/apps` or `--redirect-url-path=/app/auth/callback` to override the default values. 
Another option would be, if the CLI updates only the hostname `trash-shopping-health-inspection.trycloudflare.com` of each URL and doesn't update the paths.

### Motivation

I have to work with multiple apps (custom and public apps) on the same backend. In order to re-use the same infrastructure, I have to distinguish the different apps by their ID. I do that by simply adding `/apps/<app-id>` as prefix to all app URLs. 

1. https://trash-shopping-health-inspection.trycloudflare.com/apps/123/
2. https://trash-shopping-health-inspection.trycloudflare.com/apps/456/

These two URLs would go through the same Remix backend and the right config is loaded depending on the `<app-id>`. 

The static URL paths from the CLI make it very hard to make `app dev` work.

**Comment by gonzaloriestra** [comment]
Hi! Thanks for the suggestion, I think it makes sense. @nickwesselman what do you think?

**Comment by zirkelc** [comment]
@nickwesselman sorry to ping you. Do you think this feature would be possible?

**Comment by zirkelc** [comment]
> Hi [@zirkelc](https://github.com/zirkelc) -- Sorry for the delayed reply. I dug into the logic here a bit and confirmed that this is actually already possible in the `shopify.web.toml` with the `auth_callback_path` option. For example, if I add the last line here to my `shopify.web.toml`:
> 
> ```
> name = "remix"
> roles = ["frontend", "backend"]
> webhooks_path = "/webhooks"
> auth_callback_path = ["/hello/world","/foo/bar"]
> ```
> 
> Then when I run `shopify app dev` the URLs are updated to:
> 
> ```
> [auth]
> redirect_urls = [
>   "https://social-consortium-monroe-cleaner.trycloudflare.com/hello/world",
>   "https://social-consortium-monroe-cleaner.trycloudflare.com/foo/bar"
> ]
> ```
> 
> Does this solve your need?

Yes, but I would like the same behavior for `application_url`.  See this comment https://github.com/Shopify/cli/issues/4219#issuecomment-2232766745

```
name = "remix"
roles = ["frontend", "backend"]
webhooks_path = "/webhooks"
auth_callback_path = ["/hello/world","/foo/bar"]

# New
app_path = "/foo"
```

Then `shopify app dev` should update the application URL:

```
application_url = "https://trash-shopping-health-inspection.trycloudflare.com/foo"
```

**Comment by nickwesselman** [comment]
Hi @zirkelc -- We are open to the feature but before accepting a PR for this, the team raised a couple additional questions we'd need to answer --

1. Apps can have multiple `web.toml`, how would we resolve which to use? This isn't an issue for `auth_callback_path` since they can just be concatenated.
2. Does this affect the UI extension preview server at all? This probably requires testing -- does appending to the app path affect the expected URL for the HMR websocket connection?
  * Thinking about this more, probably not -- you could use a path in your App URL without this configuration. But it would be good to confirm.

**Comment by zirkelc** [comment]
not stale
